### PR TITLE
Fix fling velocity for precise wheel scroll

### PR DESCRIPTION
**Pull request by MatkovIvan**
Fix fling velocity for precise wheel scroll by taking into account time between events

## Testing

Manually, by pager snap behaviour

### Fixes - Desktop
- Fix fling velocity for precise wheel scroll

### CODE REVIEW

File: compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt [hunk 1]
```diff
@@ -211,6 +224,7 @@ internal class MouseWheelScrollNode(
             return
         }
         var animationState = AnimationState(0f)
+        var waitNextEventTimeMs = 0
```
Comment (igordmn) [inline compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt]: `dispatchMouseWheelScroll` becomes complex and needs refactoring (and tests) in the future. We can make it more understandable with less bugs by reducing/combining states and reading them more closely to where they are written.

File: compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt [hunk 2]
```diff
@@ -165,16 +165,29 @@ internal class MouseWheelScrollNode(
      * It's important for tests that attempt to trigger another scroll after a mouse wheel event.
      */
     private suspend fun Channel<MouseWheelScrollDelta>.busyReceive() = coroutineScope {
+        var firstFrameTimeMs: Long? = null
+        var lastFrameTimeMs: Long? = null
         val job = launch {
             while (coroutineContext.isActive) {
-                withFrameNanos { }
+                withFrameMillis { frameTimeMs ->
```
Comment (igordmn) [inline compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt]: Maybe we should use the scroll event time instead? Not sure that the frame time is the right thing to use for calculating scroll velocity

File: compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt (hunk 1 above)
Comment (MatkovIvan) [reply compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt]: Simplifying/refactoring is always welcome, but it's probably not a blocker here

File: compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt (hunk 2 above)
Comment (MatkovIvan) [reply compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt]: There is no common way to extract event time, it's platform specific. Currently it's the only source of time here

Comment (MatkovIvan) [reply compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/MouseWheelScrollable.kt]: Ah. it's in `PointerInputChange`. 🤔

**Review by igordmn — APPROVED** [review]

**Pull request merged**